lib/ansible/module_utils/network/vyos/config/lag_interfaces/lag_interfaces.py

```diff
@@ -0,0 +1,381 @@
+
+        if temp_have_members:
+            have['members'] = temp_have_members
+        if temp_want_members:
+            want['members'] = temp_want_members
+
+        commands.extend(self._add_bond_members(want, have))
+
+        if updates:
+            for key, value in iteritems(updates):
+                if value:
+                    if key == 'arp_monitor':
+                        commands.extend(
+                            self._add_arp_monitor(updates, key, want, have)
+                        )
+                    else:
+                        commands.append(self._compute_command(have['name'], key, str(value)))
+        return commands
+
+    def _render_set_commands(self, want):
+        commands = []
+        have = []
+
+        params = Lag_interfaces.params
+
+        for attrib in params:
+            value = want[attrib]
+            if value:
+                if attrib == 'arp_monitor':
+                    commands.extend(
+                        self._add_arp_monitor(want, attrib, want, have)
+                    )
+                elif attrib == 'members':
+                    commands.extend(
+                        self._add_bond_members(want, have)
+                    )
+                elif attrib != 'name':
+                    commands.append(
+                        self._compute_command(want['name'], attrib, value=str(value))
+                    )
+        return commands
+
+    def _purge_attribs(self, have):
+        commands = []
+        for item in Lag_interfaces.params:
+            if have.get(item):
+                if item == 'members':
+                    commands.extend(
+                        self._delete_bond_members(have)
+                    )
+                elif item != 'name':
+                    commands.append(
+                        self._compute_command(have['name'], attrib=item, set_flag=False)
+                    )
+        return commands
+
+    def _render_del_commands(self, want, have):
+        commands = []
+
+        params = Lag_interfaces.params
+        for attrib in params:
+            if attrib == 'members':
+                commands.extend(
+                    self._update_bond_members(attrib, want, have)
+                )
+            elif attrib == 'arp_monitor':
+                commands.extend(
+                    self._update_arp_monitor(attrib, want, have)
+                )
+            elif have.get(attrib) and not want.get(attrib):
+                commands.append(
+                    self._compute_command(have['name'], attrib, set_flag=False)
+                )
+        return commands
+
+    def _add_bond_members(self, want, have):
+        commands = []
+        diff_members = get_lst_diff_for_dicts(want, have, 'members')
+        if diff_members:
+            for key in diff_members:
+                commands.append(
+                    self._compute_command(key['member'], 'bond-group', want['name'], type='ethernet')
```

Comment:
`type` can be determined from the name of the interface.

Reply:
As discussed logic is little bit different here and only needed to pass interface in two cases .though I have renamed the variables to somewhat align with  vyos interfaces compute function .